pyuvdata/uvcal.py

```diff
@@ -231,6 +233,202 @@ def set_unknown_cal_type(self):
+                raise ValueError('Only one of antenna_nums and antenna_names can be provided.')
+
+            antenna_names = uvutils.get_iterable(antenna_names)
+            antenna_nums = []
+            for s in antenna_names:
+                if s not in self.antenna_names:
+                    raise ValueError('Antenna name {a} is not present in the antenna_names array'.format(a=s))
+                ind = np.where(np.array(self.antenna_names) == s)[0][0]
+                antenna_nums.append(self.antenna_numbers[ind])
+
+        if antenna_nums is not None:
+            antenna_nums = uvutils.get_iterable(antenna_nums)
+            history_update_string += 'antennas'
+            n_selects += 1
+
+            ant_inds = np.zeros(0, dtype=np.int)
+            for ant in antenna_nums:
+                if ant in self.ant_array:
+                    ant_inds = np.append(ant_inds, np.where(self.ant_array == ant)[0])
+                else:
+                    raise ValueError('Antenna number {a} is not present in the '
+                                     ' array'.format(a=ant))
+
+            ant_inds = list(sorted(set(list(ant_inds))))
+            self.Nants_data = len(ant_inds)
+            self.ant_array = self.ant_array[ant_inds]
+            self.flag_array = self.flag_array[ant_inds, :, :, :, :]
+            if self.cal_type == 'delay':
+                self.quality_array = self.quality_array[ant_inds, :, :, :]
+                self.delay_array = self.delay_array[ant_inds, :, :, :]
+            else:
+                self.quality_array = self.quality_array[ant_inds, :, :, :, :]
+                self.gain_array = self.gain_array[ant_inds, :, :, :, :]
+
+            if self.input_flag_array is not None:
+                self.input_flag_array = self.input_flag_array[ant_inds, :, :, :, :]
+
+        if times is not None:
+            times = uvutils.get_iterable(times)
+            if n_selects > 0:
+                history_update_string += ', times'
+            else:
+                history_update_string += 'times'
+            n_selects += 1
+
+            time_inds = np.zeros(0, dtype=np.int)
+            for jd in times:
+                if jd in self.time_array:
+                    time_inds = np.append(time_inds, np.where(self.time_array == jd)[0])
+                else:
+                    raise ValueError('Time {t} is not present in the time_array'.format(t=jd))
+
+            time_inds = list(sorted(set(list(time_inds))))
+            self.Ntimes = len(time_inds)
+            self.time_array = self.time_array[time_inds]
+            # should we adjust the time_range parameter?
+
+            time_separation = self.time_array[1:] - self.time_array[:-1]
+            if np.min(time_separation) < np.max(time_separation):
+                warnings.warn('Selected times are not evenly spaced. This '
+                              'will make it impossible to write this data out to '
+                              'some file types')
```

Comment:
Maybe change this to say it's not supported by calfits (the only data format we can write)